From 5cb6148e8e6c8d175cac45259fe37780a66b13d7 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jose.dapena@lge.com>
Date: Tue, 23 Jul 2019 11:06:48 +0200
Subject: BinaryUploadService: change parameter passing that cannot afford
 abstract class

The method UploadForDeepScanning gets a Request as parameter. But Request is an
abstract class, so GCC will not allow that declaration (polimorphycs should be
passed by reference). Use std::unique_ptr so BinaryUploadService can assume
ownership.

Bug: 819294
Change-Id: I9e8c75cc92b01abd704d9049b0421555377da5ba

diff --git a/chrome/browser/safe_browsing/download_protection/binary_upload_service.cc b/chrome/browser/safe_browsing/download_protection/binary_upload_service.cc
index 6430c89de49d..4e9048762366 100644
--- a/chrome/browser/safe_browsing/download_protection/binary_upload_service.cc
+++ b/chrome/browser/safe_browsing/download_protection/binary_upload_service.cc
@@ -10,7 +10,7 @@
 namespace safe_browsing {
 
 void BinaryUploadService::UploadForDeepScanning(
-    BinaryUploadService::Request request) {
+    std::unique_ptr<BinaryUploadService::Request> request) {
   DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
   NOTREACHED();
 }
diff --git a/chrome/browser/safe_browsing/download_protection/binary_upload_service.h b/chrome/browser/safe_browsing/download_protection/binary_upload_service.h
index d2dfd83410c3..9b6f3958e1fe 100644
--- a/chrome/browser/safe_browsing/download_protection/binary_upload_service.h
+++ b/chrome/browser/safe_browsing/download_protection/binary_upload_service.h
@@ -5,6 +5,8 @@
 #ifndef CHROME_BROWSER_SAFE_BROWSING_DOWNLOAD_PROTECTION_BINARY_UPLOAD_SERVICE_H_
 #define CHROME_BROWSER_SAFE_BROWSING_DOWNLOAD_PROTECTION_BINARY_UPLOAD_SERVICE_H_
 
+#include <memory>
+
 #include "base/callback.h"
 #include "components/safe_browsing/proto/webprotect.pb.h"
 
@@ -40,6 +42,7 @@ class BinaryUploadService {
    public:
     // |callback| will run on the UI thread.
     explicit Request(Callback callback);
+    virtual ~Request() = default;
     Request(const Request&) = delete;
     Request& operator=(const Request&) = delete;
 
@@ -67,7 +70,7 @@ class BinaryUploadService {
   // Upload the given file contents for deep scanning. The results will be
   // returned asynchronously by calling |request|'s |callback|. This must be
   // called on the UI thread.
-  void UploadForDeepScanning(Request request);
+  void UploadForDeepScanning(std::unique_ptr<Request> request);
 };
 
 }  // namespace safe_browsing
-- 
2.20.1

